src/test/java/ArrayIndexedBagTest.java

```diff
@@ -7,406 +12,513 @@
+                assertTrue(classUnderTest.remove(Integer.valueOf(10)));
+            }
+
+            @Test
+            void remove_existingElement_emptyCollection() {
+                classUnderTest.remove(Integer.valueOf(10));
+                assertEquals(new ArrayIndexedBag<Integer>(), classUnderTest);
+            }
+
+            @Test
+            void remove_nonexistentElement_throwsNoSuchElementException() {
+                assertThrows(NoSuchElementException.class, () -> classUnderTest.remove(Integer.valueOf(0)));
+            }
+
+            @Test
+            void indexOf_existingElement_returnsCorrectIndex() {
+                assertEquals(0, classUnderTest.indexOf(10));
+            }
+
+            @Test
+            void indexOf_nonexistentElement_throwsNoSuchElementException() {
+                assertThrows(NoSuchElementException.class, () -> classUnderTest.indexOf(0));
+            }
+
+            @Test
+            void contains_existingElement_returnsTrue() {
+                assertTrue(classUnderTest.contains(10));
+            }
+
+            @Test
+            void contains_nonexistentElement_returnsFalse() {
+                assertFalse(classUnderTest.contains(0));
+            }
+
+            @Test
+            void count_existingElement_returnsOne() {
+                assertEquals(1, classUnderTest.count(10));
+            }
+
+            @Test
+            void count_nonexistentElement_returnsZero() {
+                assertEquals(0, classUnderTest.count(0));
+            }
+
+            @Test
+            void isEmpty_singleton_returnsFalse() {
+                assertFalse(classUnderTest.isEmpty());
+            }
+
+            @Test
+            void size_singleton_returnsOne() {
+                assertEquals(1, classUnderTest.size());
+            }
+
+            @Test
+            void iterator_singleton_hasNextTrueOnce() {
+                Iterator<Integer> iterator = classUnderTest.iterator();
+                assertTrue(iterator.hasNext());
+                iterator.next();
+                assertFalse(iterator.hasNext());
+            }
+
+            @Test
+            void iterator_singleton_nextReturnsCorrectElements() {
+                Iterator<Integer> iterator = classUnderTest.iterator();
+                assertEquals(10, iterator.next());
+                assertThrows(NoSuchElementException.class, () -> iterator.next());
+            }
+
+            @Test
+            void toString_singleton_returnsCorrectString() {
+                assertEquals("10, ", classUnderTest.toString());
+            }
+
+            @Nested
+            @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+            class WhenMany {
+
+                @BeforeEach
+                void addMany() {
+                    classUnderTest.add(20);
+                    classUnderTest.add(30);
+                    classUnderTest.add(40);
+                    classUnderTest.add(50);
```

Comment:
Since the add that always adds to the end was used for the setting up, I did not bother having tests for it adding to the end because several tests would break if they didnt anyways.